gon/base.py

```diff
@@ -27,82 +27,9 @@ def y(self) -> Scalar:
     __repr__ = generate_repr(__init__)
 
     def __hash__(self) -> int:
-        return hash((self._x, self._y))
+        return hash((self._x, self._y, self._x + self._y, self._x - self._y))
```

Comment:
return as it was before, modify test for connection with equality using property that equality check is stronger than hash equality, they are not equivalent